File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans=new int[2];
        int n=nums.length;
        HashMap<Integer,Integer> mp = new HashMap<Integer,Integer>();
        for(int i=0;i<n;i++){
            int find = target-nums[i];
            if(mp.containsKey(find)){
                ans[0]=i;
                ans[1]=mp.get(find);
                break;
            }
            mp.put(nums[i],i);
        }
        
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            int dif = target-nums[i];
            if(mp.find(dif)!=mp.end()){
                return {i,mp[dif]};
            }
            mp[nums[i]]=i;
        }
        return {};
    }
};